shiny/Data-Ingest/server_files/ingest_workflow_svr.R

```diff
@@ -118,29 +118,31 @@ output$input_record_df <- renderPrint({Shared.data$input_record_df})
 
 
 ######### Formats Svr #############
-
-## Output List ##
-FormatRecordList <- list()
-
 output$autoname <- renderPrint({Shared.data$selected_row}) #_local
 
 ######### Mimetype Name ##################
 updateSelectizeInput(session, "MimetypeName", choices = mimetypes, server = TRUE)
 
 observeEvent(input$createFormatRecord, {
+  ## Output List ##
+  FormatRecordList <<- list()
   ## MimetypeID
   FormatRecordList$mimetypeName <- input$MimetypeName
   FormatRecordList$mimetypeID <- mimetype_sub %>% dplyr::filter(type_string %in% FormatRecordList$mimetypeName) %>% pull(id)
   
   ## Everything else
-  FormatRecordList$NewMimeType <- input$NewMimeType
-  FormatRecordList$NewFormatName <- input$NewFormatName
-  FormatRecordList$HeaderBoolean <- input$HeaderBoolean
-  FormatRecordList$SkipLines <- input$SkipLines #This should appear only if header = TRUE
-  FormatRecordList$FormatNotes <- input$FormatNotes
+  FormatRecordList$NewMimeType <- "text/csv" #input$NewMimeType
+  FormatRecordList$NewmimetypeID <- mimetype_sub %>% dplyr::filter(type_string %in% FormatRecordList$NewMimeType) %>% pull(id)
+  FormatRecordList$NewFormatName <- "Test_Format_Name_99" #input$NewFormatName
+  FormatRecordList$HeaderBoolean <- "TRUE" #ifelse((input$HeaderBoolean == "Yes"), "TRUE", "FALSE")
+  FormatRecordList$SkipLines <- "3" #input$SkipLines #This should appear only if header = TRUE
+  FormatRecordList$FormatNotes <- "These are some test notes" #input$FormatNotes
   
   output$FormatRecordOut <- renderPrint({print(FormatRecordList)})
   
+  dbfile.format.variable.insert(header = FormatRecordList$HeaderBoolean, skip = FormatRecordList$SkipLines, mimetype_id = FormatRecordList$NewmimetypeID,
+                                format_notes = FormatRecordList$FormatNotes, format_name = FormatRecordList$NewFormatName, con = bety$con, format_variables = FALSE)
+
```

Comment:
Using `db_merge_into` as discussed above, you may not actually need `dbfile.format.variable.insert` at all. You may just be able to get away with constructing `FormatRecordList` as a `data.frame` instead (with as many rows as you want -- it sounds like you could be adding multiple variables at once, which `db_merge_into` is specifically designed to handle) and then just insert it all at once with a single `db_merge_into` statement. A nice advantage of `db_merge_into` in the context of Shiny is that, by design, it will never insert duplicate rows (at least, with a well-specified `by` argument), so it can be more-or-less safely called over-and-over in a single session (though there is a bit of computational overhead, so I would avoid a design that calls it hundreds of times).

@mdietze Was this more-or-less what you had in mind?

Reply:
I think it's still worth having the function, even if it only ends up being 2-3 lines, so that it doesn't have to be recreated in the future. I had asked @LiamBurke24 to make this a function, rather than just having the insert code directly in the shiny app (original version) precisely because I find going through BETY's current multi-step process of creating the format, and then editing that record to associate variables with the format, to be a pain. The fact that it's taking 4 of us having a long discussion to sort out how to do this correctly implies that not having a function could lead to mistakes by others.

Additional point -- I wonder if this function should return the full format-variable object that @bcow has query.format.vars returns rather than just the format ID? I don't think anyone is going to need to create a format and then immediately use it in `load_data`, but it's worth considering.

Reply:
@mdietze @ashiklom So we're all on the same page. I am going to use `db_merge_into` to rewrite the `dbfile.format.variable` function (which will shortly be renamed to something with out `dbfile` in the name). 

I would also like to clarify what we would like the function to be returning. SHould I start with the `format ID` and then proceed to the full format-variable object if need be/ time permitting?

Finally, @ashiklom based on what you're saying, db_merge_into looks like a perfect fit for this application. 

P.S. where should I be storing this file? In the `dbfiles.R` document? Or somewhere else?

Reply:
* yes, `insert.format.vars` is my vote for name
* returning format ID is sufficient for now.
* put it in it's own file

Reply:
@mdietze Awesome. Thanks!

Reply:
Follow-up question: where should I put this new function? Is `pecan/base/db/R/` ok?

Reply:
Sorry, yes it should definitely be in the db package! `pecan/base/db/R/`

Reply:
Perfect. Thanks!